### PR TITLE
Rename `databricks-resources` folder to `resources`

### PR DESCRIPTION
**Pull request by zhe-db**
Rename databricks-resources folder to resources

Test run of staging action: https://github.com/zhe-db/my-test-project/actions/runs/6218036380/job/16874018452

### CODE REVIEW

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/training/notebooks/TrainWithMLflowRecipes.py.tmpl [hunk 1]
```diff
@@ -4,7 +4,7 @@
 ##
 # This notebook runs the MLflow Regression Recipe to train and registers an MLflow model in the model registry.
 # It is configured and can be executed as the "Train" task in the model_training_job workflow defined under
-# ``{{template `project_name_alphanumeric_underscore` .}}/databricks-resources/model-workflow-resource.yml``
+# ``{{template `project_name_alphanumeric_underscore` .}}/esources/model-workflow-resource.yml``
```
Comment (mingyu89) [inline template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/training/notebooks/TrainWithMLflowRecipes.py.tmpl:7]: resources

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/databricks.yml.tmpl [hunk 2]
```diff
@@ -15,18 +15,18 @@ variables:
 include:
```
Comment (mingyu89) [inline template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/databricks.yml.tmpl:15]: As the comment suggested, let's just use `- ./resources/*`. Maybe merge the comments into one about what's included in the `resources` directory.

Comment (zhe-db) [reply template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/databricks.yml.tmpl:15]: Do we want to include feature engineering workflow by default (I guess it is fine as it is not being used).

Comment (mingyu89) [reply template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/databricks.yml.tmpl:15]: The feature engineering workflow will not present in the directory if it's not used.

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/databricks.yml.tmpl [hunk 3]
```diff
@@ -13,26 +13,16 @@ variables:
...
-  # Defines workflow for feature engineering
-  - ./databricks-resources/feature-engineering-workflow-resource.yml{{ end }}
-  # Defines workflow for scheduled batch inference
-  - ./databricks-resources/batch-inference-workflow-resource.yml
-  # Defines workflow for data monitoring, metric refresh, alerts and triggering retraining
-  - ./databricks-resources/monitoring-workflow-resource.yml
-
+  # Resources folder contains ML artifact resources for the ml project that defines model and experiment
+  # And workflows resources for the ml project including model training -> validation -> deployment, 
+  # feature engineerng, batch inference, data monitoring, metric refresh, alerts and triggering retraining
```
Comment (mingyu89) [inline template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/databricks.yml.tmpl:18]: Let's add back the condition for feature engineering in the comment
```
{{- if (eq .input_include_feature_store `yes`) }}feature engineerng, {{ end }}
```

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/databricks.yml.tmpl [hunk 4]
```diff
@@ -13,26 +13,16 @@ variables:
...
+  - ./resources/*
 
 # Deployment Target specific values for workspace
 targets:
   dev:
     default: true
+    mode: development
```
Comment (mingyu89) [inline template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/databricks.yml.tmpl:25]: Let's remove the mode change and wait until staging is supported.

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/databricks.yml.tmpl [hunk 5]
```diff
@@ -13,26 +13,16 @@ variables:
...
-  - ./databricks-resources/feature-engineering-workflow-resource.yml{{ end }}
-  # Defines workflow for scheduled batch inference
-  - ./databricks-resources/batch-inference-workflow-resource.yml
-  # Defines workflow for data monitoring, metric refresh, alerts and triggering retraining
-  - ./databricks-resources/monitoring-workflow-resource.yml
-
+  # Resources folder contains ML artifact resources for the ml project that defines model and experiment
+  # And workflows resources for the ml project including model training -> validation -> deployment, 
+  # feature engineerng, batch inference, data monitoring, metric refresh, alerts and triggering retraining
+  - ./resources/*
```
Comment (mingyu89) [inline template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/databricks.yml.tmpl:19]: There are test failures. Looks like it also tried to import the README.md as yaml file and failed.
Could you try `./resources/*.yml`?
If it doesn't work, we can revert back to specifying each file.

File: template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/databricks.yml.tmpl [hunk 6]
```diff
@@ -42,6 +32,7 @@ targets:
       host: {{template `databricks_staging_workspace_host` .}}
 
   prod:
+    mode: production
```
Comment (mingyu89) [inline template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/databricks.yml.tmpl:35]: same here

**Review by mingyu89 — APPROVED** [review]
LGTM. Please do a sanity check to make sure the resources in `resources` will be included during bundle validation and deployment.

**Comment by zhe-db** [comment]
> LGTM. Please do a sanity check to make sure the resources in `resources` will be included during bundle validation and deployment.

CleanShot 2023-09-26 at 10 36 07@2x
CleanShot 2023-09-26 at 10 36 22@2x

**Pull request merged**